### PR TITLE
Removed some D2L-ness and setup to publish to npmjs.org

### PR DESCRIPTION
**Pull request by jkehres**
Major version bump. See `CHANGELOG.md` for complete details.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Changelog
+
+## 4.0.0
+### Breaking Changes
+- Renamed executable command from `license-checker-ci` to `d2l-license-checker`.
```
Comment (jkehres) [inline CHANGELOG.md:5]: Repo was renamed previously but command wasn't. Now it matches repo name.

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+- Renamed special exemption identifier from `D2L-Open-Source-Special-Exemption` to `Special-Exemption`.
```
Comment (jkehres) [inline CHANGELOG.md:6]: Trying to make this tool more likely to be used outside D2L.

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+- Changed specifying a custom config file from a positional argument to `--config-file` option.
+
+### Enhancements
+- Added `--no-d2l` option to bypass built-in defaults that are D2L-specific.
+- Allow specifying `"acceptedLicenses"` in config file when `--no-d2l` is used.
```
Comment (jkehres) [inline CHANGELOG.md:11]: Again trying to make this tool be less D2L-centric when desired to entice non-D2L users.

File: README.md [hunk 4]
```diff
@@ -33,21 +31,23 @@ Simple tool to continuously check licenses of all npm dependencies in a project.
...
 
-  - `Public-Domain`: identifier for public domain code (not supported by SPDX)
-  - `Project-Owner`: identifier indicating that you own this package and that its license can be ignored (doesn't need to be added to `acceptedlicenses`)
-  - `D2L-Open-Source-Special-Exemption (license-name)`: identifier indicating that although `license-name` is not a D2L-approved open source license, its use has been granted a special-exemption for this project.
+* `"manualOverrides"`: Object where each key is a package name and version (see above example), and the value is a valid SPDX ID. The version number can be a semver expression. You can use this to manually specify the license of a package for which the license is not specified in its `package.json` file or where an invalid SPDX ID is used. The default config is a set of overrides for packages used by D2L. If `--no-d2l` is specified, the default is an empty object.
 
-* `"acceptedScopes"`: List of [NPM scopes](https://docs.npmjs.com/misc/scope) that should always be accepted. This is convenient if your team uses its own scoped registry. Do not include the `@` or `/` characters. The default config is `["d2l"]`.
```
Comment (jkehres) [inline README.md:50]: Moved this up to line 42.

**Comment by bearfriend** [comment]
Was the opt-in to d2l defaults intentionally chosen over opt-out?

File: test/testBin.js [hunk 5]
```diff
@@ -50,54 +57,93 @@ const makeTestPath = (proj) => path.join(dataDir, proj);
...
+	});
+
+	it('should allow a custom file', () => {
+		assert.equal(checkProject({
+			projectPath: makeTestPath('proj-ok-custom-file'),
+			configFile: 'myfile.json'
+		}), 0);
+	});
+
+	it.only('should allow acceptedLicenses for no-d2l', () => {
```
Comment (AKobets) [inline test/testBin.js:139]: remove .only

**Comment by jkehres** [comment]
> Was the opt-out of d2l defaults intentionally chosen over opt-in?

Yes. I thought most consumers would be D2Lers so best to make it setup for them by default. If the `--no-d2l` option has little value (this tool is called `d2l-license-checker` so it's already predisposed to being a D2L-specific tool), then I can remove it. It's not something we at D2L need and the naming/nature of this tool may mean no one outside D2L wants to use it. In the long run, it would be nice to make this tool totally D2L independent and instead make it easy to import a D2L-specific config that is managed separately.

File: test/testBin.js (hunk 5 above)
Comment (jkehres) [reply test/testBin.js]: Done. Good catch.

File: README.md [hunk 6]
```diff
@@ -1,21 +1,19 @@
...
-Simple tool to continuously check licenses of all npm dependencies in a project. Can be added to a test suite / CI to get a warning about packages not meeting predefined license requirements. This is basically a wrapper around [`davglass/license-checker`](https://github.com/davglass/license-checker)
+Simple tool to check licenses of all npm dependencies in a project against an approved set of licenses. Can be added to a test suite / CI to get a warning about packages not meeting predefined license requirements. This is basically a wrapper around [`davglass/license-checker`](https://github.com/davglass/license-checker)
 
 ## How to use
 
-1. Add this package as a dev-requirement:
+1. Add this package as a development dependency:
 
-    `node install --save-dev @d2l/d2l-license-checker`
+    `node install --save-dev d2l-license-checker`
```
Comment (AKobets) [inline README.md:11]: should it be `npm install ...`

Comment (jkehres) [reply README.md]: Fixed.

**Comment by jkehres** [comment]
@AKobets what do you think about making this tool less D2L-specific? Should I revert that part of the PR and just focus on the automated publishing to npmjs.org and fixing the docs? I kind of feel like what I've done is a half hearted attempt to make this tool more general but maybe it should be an all or nothing approach - either this tool is totally D2L independent or it doesn't apologize about being blatantly D2L-specific.

**Pull request closed without merging**